Clean-up and minor optimizations WIP

mrpgraae opened the pull request:
Initial clean-up of the code and PEP 8 compliance. Also minor optimizations, if these are also considered code cleaning.

mrpgraae committed:
flake it

mrpgraae committed:
add missing newline at end of .gitignore

mrpgraae committed:
:lipstick:

mrpgraae committed:
:lipstick:

mrpgraae committed:
fix config parser

mrpgraae committed:
Add tests

mrpgraae committed:
ignore output folder

mrpgraae committed:
don't ignore exampletableoutput.txt

radosuav commented on pyTSEB/TSEBConfigFileInterface.py:
```diff
@@ -175,10 +175,12 @@ def get_data(self, config_data, is_image):
+                        self.params[var_name] = str(config_data[var_name]).strip('"')
+                elif var_name == "correct_LST":
+                    if self.params["model"] == "disTSEB":
+                        self.params[var_name] = int(config_data[var_name])
```
Maybe it's nicer like in https://github.com/hectornieto/pyTSEB/commit/ed38e4519c76f920a5cb7f052eee94cfad9e166a

mrpgraae replied:
I might also give the config parsing an overhaul, so I'll keep it as is for now.

radosuav commented on pyTSEB/net_radiation.py:
```diff
@@ -115,17 +113,17 @@ def calc_difuse_ratio(S_dn, sza, press=1013.25, SOLAR_CONSTANT=1320):
 
 
-def calc_emiss_atm(ea, T_A_K):
+def calc_emiss_atm(ea, t_a_k):
```
I know that by convention all capital variables refer to constants, but in this case it refers to a symbol from model equation. So for consistency with equations in published papers and in other places in the code (e.g. T_R_K - radiometric temperature in Kelvin, z_T - temperature measurement height) I would prefer if we leave it as T_A_K. There are plenty of other variables with similar capital names (e.g. H, LE, R_A) which I would also suggest to leave as they are.

mrpgraae replied:
It's a bit confusing, given that all capital letters means top-level constant and most syntax highlighters will also color the variables as such, but if you believe it will hurt code readability more if we change it, we should keep it as all capital letters.

mrpgraae committed:
more :lipstick: and slight optimizations

mrpgraae committed:
flake 'em all

radosuav committed:
Update TSEB_PT outputs which will be used in unit tests.

They were derived with un-optimised code and will be used to ensure that optimisations do not change the model results.

mrpgraae committed:
Merge pull request #2 from DHI-GRAS/tests

Merge new test data into clean-up branch

mrpgraae committed:
overhaul config parsing

mrpgraae committed:
minor re-factoring

mrpgraae committed:
more clean-up

mrpgraae committed:
finish config parser

mrpgraae committed:
add a tolerance to test

mrpgraae committed:
fix inline comment

mrpgraae commented:
@radosuav I'm going to merge this now and make a new PR, just so we don't end up with branches that diverge too much. Should I also make a PR to merge it into Hectors code? Our fork has probably already diverged a lot from his, so in order to avoid too many merge conflicts, it might be a good idea.

radosuav commented:
Sounds like a good plan, both for merging into our master and Hectors code.
